pallets/task/src/tests.rs

```diff
@@ -937,3 +937,83 @@ fn block_time_is_added_when_task_is_updated() {
 
 	})
 }
+
+#[test]
+fn test_multiple_tasks_and_reserveamounts() {
+	new_test_ext().execute_with( || {
+
+		assert_ok!(Profile::create_profile(Origin::signed(1), username(), interests(), HOURS, Some(additional_info())));
+
+		//create 2 tasks of budgets 7 and 10
+		assert_ok!(Task::create_task(Origin::signed(1), title(), spec(), BUDGET, get_deadline(), attachments(), keywords()));
+		assert_ok!(Task::create_task(Origin::signed(1), title2(), spec2(), BUDGET2, get_deadline(), attachments2(), keywords2()));
+
+		// Assert that the reserved balances add up
+		assert_eq!(Balances::reserved_balance(&1), BUDGET + BUDGET2);
+
+		//swap around budgets
+		let hash = Task::tasks_owned(1)[0];
+		assert_ok!(Task::update_task(Origin::signed(1), hash, title(), spec(), BUDGET2, get_deadline(), attachments(), keywords()));
+
+		assert_eq!(Balances::reserved_balance(&1), BUDGET2 + BUDGET2);
+
+		//let hash = Task::tasks_owned(1)[1];
+		assert_ok!(Task::update_task(Origin::signed(1), hash, title2(), spec2(), BUDGET, get_deadline(), attachments2(), keywords2()));
+
+		assert_eq!(Balances::reserved_balance(&1), BUDGET2 + BUDGET);
+	})
+
+}
+
+#[test]
+fn test_create_not_enough_funds_to_reserve() {
+	new_test_ext().execute_with( || {
+		assert_ok!(Profile::create_profile(Origin::signed(1), username(), interests(), HOURS, Some(additional_info())));
+		
+		//Create a task with more tokens than the signer has
+		let res = Task::create_task(Origin::signed(1), title(), spec2(), Balances::free_balance(&1) + 1000, get_deadline(), attachments(), keywords());
+		
+		//hack to make work
+		if let Err(n) = res {
+			assert_eq!(n.error, Error::<Test>::NotEnoughBalance.into());	
+		}
+
+	})
+}
+#[test]
+fn test_update_not_enough_funds_to_reserve() {
+	new_test_ext().execute_with( || {
+		assert_ok!(Profile::create_profile(Origin::signed(1), username(), interests(), HOURS, Some(additional_info())));
+		
+		//Create task that should be ok
+		assert_ok!(Task::create_task(Origin::signed(1), title(), spec2(), Balances::free_balance(&1) - 1000, get_deadline(), attachments(), keywords()));
+		
+		let hash = Task::tasks_owned(1)[0];
+
+		//update that task with a balance more than signer has
+		let res = Task::update_task(Origin::signed(1), hash, title2(), spec2(), Balances::free_balance(&1) + 1000, get_deadline(), attachments2(), keywords2());
+		if let Err(n) = res {
+			assert_eq!(n.error, Error::<Test>::NotEnoughBalance.into());	
+		}
+
+	})
+}	
+
+#[test]
+fn test_create_two_task_not_enough_balance() {
```

Comment:
Maybe test that after acceptance the budget is transferred to the volunteer?

Reply:
Done already in test 
```fn check_balance_after_complete_task()``` line 282